### PR TITLE
8274730: AArch64: AES/GCM acceleration is broken by the fix for JDK-8273297

### PR DESCRIPTION
**Pull request by theRealAph**
The recent AES/GCM acceleration on AArch64 was broken by https://bugs.openjdk.java.net/browse/JDK-8273297 . This was entirely expected, and I approved the patch, but now we must make AArch64 acceleration work again. 
The only significant change from the point of view of this patch is that one argument was added to the call to the intrinsic, and that argument caused another argument to spill onto the stack.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8274730](https://bugs.openjdk.java.net/browse/JDK-8274730): AArch64: AES/GCM acceleration is broken by the fix for JDK-8273297

### Reviewers
 * [Andrew Dinn](https://openjdk.java.net/census#adinn) (@adinn - **Reviewer**) ⚠️ Review applies to adfcc75cd34425b020f4cd868a628491c7fc6947
 * [Roland Westrelin](https://openjdk.java.net/census#roland) (@rwestrel - **Reviewer**) ⚠️ Review applies to adfcc75cd34425b020f4cd868a628491c7fc6947
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5819/head:pull/5819` \
`$ git checkout pull/5819`

Update a local copy of the PR: \
`$ git checkout pull/5819` \
`$ git pull https://git.openjdk.java.net/jdk pull/5819/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5819`

View PR using the GUI difftool: \
`$ git pr show -t 5819`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5819.diff">https://git.openjdk.java.net/jdk/pull/5819.diff</a>

</details>

### CODE REVIEW

**Review by adinn — APPROVED** [review]
Changes look good.

**Review by rwestrel — APPROVED** [review]
c2 change looks good to me.

File: src/hotspot/share/opto/library_call.cpp [hunk 1]
```diff
@@ -6794,14 +6794,20 @@ bool LibraryCallKit::inline_galoisCounterMode_AESCrypt() {
...
-  Node* subkeyHtbl_48_entries = new_array(klass_node, intcon(Matcher::htbl_entries), 0);
-  if (subkeyHtbl_48_entries == NULL) return false;
-
-  Node* subkeyHtbl_48_entries_start = array_element_address(subkeyHtbl_48_entries, intcon(0), T_LONG);
+  Node* subkeyHtbl_48_entries_start;
+  if (Matcher::htbl_entries != 0) {
+    Node* subkeyHtbl_48_entries = new_array(klass_node, intcon(Matcher::htbl_entries), 0);
+    if (subkeyHtbl_48_entries == NULL) return false;
+    subkeyHtbl_48_entries_start
+      = array_element_address(subkeyHtbl_48_entries, intcon(0), T_LONG);
```
Comment (vnkozlov) [inline src/hotspot/share/opto/library_call.cpp:6806]: Please, don't split such line.

File: src/hotspot/share/opto/library_call.cpp [hunk 2]
```diff
@@ -6794,14 +6794,20 @@ bool LibraryCallKit::inline_galoisCounterMode_AESCrypt() {
   ciKlass* klass = ciTypeArrayKlass::make(T_LONG);
   Node* klass_node = makecon(TypeKlassPtr::make(klass));
 
-  // htbl entries is set to 96 only fox x86-64
+  // Does this target support this intrinsic?
   if (Matcher::htbl_entries == -1) return false;
 
   // new array to hold 48 computed htbl entries
```
Comment (vnkozlov) [inline src/hotspot/share/opto/library_call.cpp:6800]: Move this comment under `(Matcher::htbl_entries != 0)` check.

**Review by vnkozlov — COMMENTED** [review]
Looks good with just 2 small comments.
I will test changes and approve it after that.

**Review by vnkozlov — APPROVED** [review]
Good. And my testing passed.

**Pull request closed without merging**